src/main/java/uk/gov/hmcts/darts/common/service/impl/AutomatedTaskServiceImpl.java

```diff
@@ -0,0 +1,162 @@
+                throw new DartsApiException(INVALID_CRON_EXPRESSION);
+            }
+        }  else {
+            log.error("Unable to update the task {} with cron expression {}", taskName, cronExpression);
+            throw new DartsApiException(FAILED_TO_FIND_AUTOMATED_TASK);
+        }
+    }
+
+    @Override
+    public void loadTaskByName(String taskName) {
+        switch (taskName) {
+            case ProcessDailyListAutomatedTask.PROCESS_DAILY_LIST_TASK_NAME -> loadProcessDailyListAutomatedTask();
+            default -> throw new DartsApiException(FAILED_TO_FIND_AUTOMATED_TASK);
+        }
+    }
+
+    public void loadProcessDailyListAutomatedTask() {
+        ProcessDailyListAutomatedTask processDailyListAutomatedTask = new ProcessDailyListAutomatedTask();
+        Trigger trigger = createAutomatedTaskTrigger(processDailyListAutomatedTask);
+        taskScheduler.schedule(processDailyListAutomatedTask, trigger);
+    }
+
+    private void loadProcessDailyListAutomatedTask(ScheduledTaskRegistrar taskRegistrar) {
+        ProcessDailyListAutomatedTask processDailyListAutomatedTask = new ProcessDailyListAutomatedTask();
+        Trigger trigger = createAutomatedTaskTrigger(processDailyListAutomatedTask);
+        taskRegistrar.addTriggerTask(processDailyListAutomatedTask, trigger);
+    }
+
+    @Override
+    public boolean cancelAutomatedTask(String taskName, boolean mayInterruptIfRunning) {
+        boolean result = false;
+        Set<ScheduledTask> scheduledTasks = taskHolder.getScheduledTasks();
+        for (ScheduledTask scheduledTask : scheduledTasks) {
+            Task task = scheduledTask.getTask();
+            if (task instanceof TriggerTask) {
+                TriggerTask triggerTask = (TriggerTask) task;
+                if (cancelTriggerTask(taskName, scheduledTask, triggerTask, mayInterruptIfRunning)) {
+                    result = true;
+                    break;
+                }
+            }
+        }
+        return result;
+    }
+
+    private Trigger createAutomatedTaskTrigger(AutomatedTask automatedTask) {
+        Trigger trigger = createCronTrigger(automatedTask);
+        taskTriggers.put(automatedTask.getTaskName(), trigger);
+        return trigger;
+    }
+
+    private boolean cancelTriggerTask(String taskName, ScheduledTask scheduledTask, TriggerTask triggerTask, boolean mayInterruptIfRunning) {
+        boolean result = false;
+        if (triggerTask.getRunnable() instanceof AutomatedTask) {
+            AutomatedTask automatedTask = (AutomatedTask) triggerTask.getRunnable();
+            if (automatedTask.getTaskName().equals(taskName)) {
+                log.info("About to cancel task: " + taskName);
+                scheduledTask.cancel(mayInterruptIfRunning);
+                result = true;
+                taskTriggers.remove(taskName);
+            }
+        }
+        return result;
+    }
```

Comment:
To improve readability, suggest eliminating the boolean flags by simply returning early, e.g.:

```java
    private boolean cancelTriggerTask(String taskName, ScheduledTask scheduledTask, TriggerTask triggerTask, boolean mayInterruptIfRunning) {
        if (triggerTask.getRunnable() instanceof AutomatedTask) {
            AutomatedTask automatedTask = (AutomatedTask) triggerTask.getRunnable();
            if (automatedTask.getTaskName().equals(taskName)) {
                log.info("About to cancel task: " + taskName);
                scheduledTask.cancel(mayInterruptIfRunning);
                taskTriggers.remove(taskName);
                return true;
            }
        }
        return false;
    }
```